src/libsyntax/ext/tt/macro_rules.rs

```diff
@@ -358,6 +358,10 @@ pub fn compile<'cx>(cx: &'cx mut ExtCtxt,
         valid &= check_rhs(cx, rhs);
     }
 
+    for lhs in &lhses {
+        valid &= check_lhs_no_empty_seq(cx, &[lhs.clone()])
+    }
```

Comment:
The `for` loop was mainly used to use the same style as the loop of the `rhses` directly above.

Also `.iter().all()` would abort early, I think, which means that not all `lhses` would be checked, ie. not all errors we could report would be reported.